### PR TITLE
AV-164034: Updating VMWare-ALB-SDK to version 22.1.2

### PR DESCRIPTION
**Pull request by akshayhavile**
Fixing issue: AV-164034
@abhishekbsingh : I have update few infra related files. Please check its validity.

Testing: Created EVH secure ingress with controller version 22.1.3. EVH VS created.

### CODE REVIEW

**Review by DixitAakash — DISMISSED** [review]
LGTM

File: ako-infra/avirest/handle_netinfo.go [hunk 1]
```diff
@@ -345,9 +345,28 @@ func addNetworkInIPAM() {
 	}
 	netName := lib.GetVCFNetworkName()
 	networkRef := "/api/network/?name=" + netName
-
-	if !utils.HasElem(ipam.InternalProfile.UsableNetworkRefs, networkRef) {
-		ipam.InternalProfile.UsableNetworkRefs = append(ipam.InternalProfile.UsableNetworkRefs, networkRef)
+	// ipam.InternalProfile.UsableNetworkRefs field is deprecated in 20.1.3
```
Comment (apalsule) [inline ako-infra/avirest/handle_netinfo.go:348]: @abhishekbsingh can you look into these changes

File: ako-infra/ingestion/avi_infra_controller.go [hunk 2]
```diff
@@ -103,7 +103,9 @@ func (a *AviControllerInfra) DeriveCloudNameAndSEGroupTmpl(tz string) (error, st
 		}
 		vtype := *cloud.Vtype
 		if vtype == lib.CLOUD_NSXT && cloud.NsxtConfiguration != nil {
-			if cloud.NsxtConfiguration.TransportZone != nil && *cloud.NsxtConfiguration.TransportZone == tz {
+			//cloud.NsxtConfiguration.TransportZone field is deprecated in 20.1.5
```
Comment (apalsule) [inline ako-infra/ingestion/avi_infra_controller.go:106]: @abhishekbsingh

**Review by swathinsankaran — DISMISSED** [review]
LGTM

**Comment by swathinsankaran** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Review by swathinsankaran — DISMISSED** [review]

**Review by DixitAakash — DISMISSED** [review]
LGTM

File: ako-infra/avirest/handle_netinfo.go [hunk 3]
```diff
@@ -345,9 +345,28 @@ func addNetworkInIPAM() {
...
+			break
+		}
+	}
+	//TODO: Check with Abishek
+	// Problem with this field is: it is applicable in Enterprise license only.
+	if !networkRefFound {
+		newUsableNetwork := &models.IPAMUsableNetwork{
+			NwRef: &networkRef,
+		}
+		newUsableNetwork.NwRef = &networkRef
```
Comment (abhishekbsingh) [inline ako-infra/avirest/handle_netinfo.go:368]: repeated assignment

File: ako-infra/ingestion/avi_infra_controller.go (hunk 2 above)
Comment (abhishekbsingh) [reply ako-infra/ingestion/avi_infra_controller.go:106]: AKO-Infra changes have already been covered in the PR https://github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pull/955/files, we can remove these changes from the PR here

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
@abhishekbsingh: I have commented out all avi-infra code which was causing conflict. Please work on it as part of your PRs.

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Review by swathinsankaran — APPROVED** [review]

**Review by abhishekbsingh — APPROVED** [review]

**Pull request merged**